examples/cluster/plot_inductive_clustering.py

```diff
@@ -0,0 +1,98 @@
+"""
+==============================================
+Inductive Clustering
+==============================================
+
+Clustering is expensive, especially when our dataset contains millions of
+datapoints. Recomputing the clusters everytime we receive some new data
+is thus in many cases, intractable. With more data, there is also the
+possibility of degrading the previous clustering.
+One solution to this problem, is to first infer the target classes using
+some unsupervised learning algorithm and then fit a classifier on the
+inferred targets, treating it as a supervised problem.
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+from sklearn.base import BaseEstimator, clone
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.datasets import make_blobs
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+
+N_SAMPLES = 5000
+RANDOM_STATE = 42
+
+
+class InductiveClusterer(BaseEstimator):
+    def __init__(self, clusterer, classifier):
+        self.clusterer = clusterer
+        self.classifier = classifier
+
+    def fit(self, X, y=None):
+        self.clusterer_ = clone(self.clusterer)
+        self.classifier_ = clone(self.classifier)
+        y = self.clusterer_.fit_predict(X)
+        self.classifier_.fit(X, y)
+        return self
+
+    @if_delegate_has_method(delegate='classifier')
```

Comment:
should be with an underscore: `@if_delegate_has_method(delegate='classifier_')`

Reply:
Done